src/main/java/io/logz/jmx2graphite/Jmx2GraphiteConfiguration.java

```diff
@@ -39,22 +37,31 @@
         MBEAN_PLATFORM
     }
 
-    public Jmx2GraphiteConfiguration(Config config, MetricClientType metricClientType) throws IllegalConfiguration {
-
-        this.metricClientType = metricClientType;
+    private class Graphite {
+        public String hostname;
+        public int port;
+    }
 
+    public Jmx2GraphiteConfiguration(Config config) throws IllegalConfiguration {
         if (config.hasPath("service.host")) {
             serviceHost = config.getString("service.host");
         }
 
+        if (config.hasPath("service.poller.jolokia")) {
+            metricClientType = MetricClientType.JOLOKIA;
+        }
+        else if (config.hasPath("service.poller.mbean-direct")) {
+            metricClientType = MetricClientType.MBEAN_PLATFORM;
+        }
+
         if (this.metricClientType == MetricClientType.JOLOKIA) {
-            jolokiaUrl = config.getString("service.jolokiaUrl");
+            jolokiaFullUrl = config.getString("service.jolokiaFullUrl");
```

Comment:
Don't you reckon that if you have a section where you specify which type of client it contain its own config? Maybe should be "service.poller.jolokia.fullUrl" ?